lib/pbench/test/functional/server/test_put.py

```diff
@@ -35,44 +39,61 @@ def test_upload_all(self, server_client: PbenchServerClient, login_user):
+                f"Unexpected HTTP error, url = {exc.response.url}, status code = {exc.response.status_code}, text = {exc.response.text!r}"
+            )
 
     @staticmethod
     def check_indexed(server_client: PbenchServerClient, datasets):
         indexed = []
         not_indexed = []
-        for dataset in datasets:
-            print(f"\t... on {dataset.metadata['server.tarball-path']}")
-            metadata = server_client.get_metadata(
-                dataset.resource_id, ["dataset.state", "server.status"]
+        try:
+            for dataset in datasets:
+                print(f"\t... on {dataset.metadata['server.tarball-path']}")
+                metadata = server_client.get_metadata(
+                    dataset.resource_id, ["dataset.state", "server.status"]
+                )
+                state = metadata["dataset.state"]
+                status = metadata["server.status"]
+                stats = set(status.keys()) if status else set()
+                if state == "Indexed" and {"unpack", "index"} <= stats:
+                    # Don't wait for backup, and don't fail if we haven't as
+                    # it's completely independent from unpack/index; but if we
+                    # have backed up, check that the status was successful.
+                    if "backup" in stats:
+                        assert status["backup"] == "ok"
+                    assert status["unpack"] == "ok"
+                    assert status["index"] == "ok"
+                    indexed.append(dataset)
+                else:
+                    not_indexed.append(dataset)
+        except HTTPError as exc:
+            pytest.fail(
+                f"Unexpected HTTP error, url = {exc.response.url}, status code = {exc.response.status_code}, text = {exc.response.text!r}"
             )
-            state = metadata["dataset.state"]
-            status = metadata["server.status"]
-            stats = set(status.keys()) if status else set()
-            if state == "Indexed" and {"unpack", "index"} <= stats:
-                # Don't wait for backup, and don't fail if we haven't as
-                # it's completely independent from unpack/index; but if we
-                # have backed up, check that the status was successful.
-                if "backup" in stats:
-                    assert status["backup"] == "ok"
-                assert status["unpack"] == "ok"
-                assert status["index"] == "ok"
-                indexed.append(dataset)
-            else:
-                not_indexed.append(dataset)
         return not_indexed, indexed
 
     def test_index_all(self, server_client: PbenchServerClient, login_user):
         """Wait for datasets to reach the "Indexed" state, and ensure that the
         state and metadata look good
         """
+        tarball_names_l = []
+        for t in TARBALL_DIR.glob("*.tar.xz"):
+            tarball_names_l.append(t.name)
+        tarball_names = frozenset(tarball_names_l)
```

Comment:
Why aren't you relying on `self.tarballs` here as above?

If the intent is to break the implicit dependencies between the tests, that's "interesting" but it seems kinda pointless to run just `test_index_all` checking only datasets we just uploaded if we didn't just upload any.

And, at the very least, this should be factored out into a common method that could be reused in each of the test cases.